py7zr/py7zr.py

```diff
@@ -340,41 +340,49 @@ def __init__(
         # check invalid mode.
         if mode not in ("r", "w", "x", "a"):
             raise ValueError("ZipFile requires mode 'r', 'w', 'x', or 'a'")
+
+        # Check if we were passed a file-like object or not
+        if isinstance(file, os.PathLike):
+            file = os.fspath(file)
+
+            # check if it's a non existent file opened in append mode.
+            if mode == "a" and not os.path.isfile(file):
+                # Nothing to append because file doesn't exist, write instead.
+                mode = "w"
```

Comment:
We may not require to check a file existence here. I think the case is handled by zipfile way. Isn't it?

Reply:
Nope. Opening with `a` does create the file but it still errors in `_check_7zfile` because the created file is missing the magic_7z.
https://github.com/miurahr/py7zr/blob/370463069a5a11bce4d7f57d82c49d67e29a87da/py7zr/py7zr.py#L404
https://github.com/miurahr/py7zr/blob/370463069a5a11bce4d7f57d82c49d67e29a87da/py7zr/py7zr.py#L428-L430
https://github.com/miurahr/py7zr/blob/370463069a5a11bce4d7f57d82c49d67e29a87da/py7zr/py7zr.py#L778-L781

Reply:
I just checked ZipFile's code and it also has special handling for append. It's done inside the append `if` statement so maybe I can move our check there too
https://github.com/python/cpython/blob/f2cc8ad989ba1039f69b62f92c2eb700d142e285/Lib/zipfile/__init__.py#L1366-L1379

Reply:
Thank you for explanation. I understand that we need to handle "a" to be special. I agree the your proposal.